File: common/src/main/java/io/github/notenoughupdates/moulconfig/common/IItemStack.java
package io.github.notenoughupdates.moulconfig.common;

import io.github.notenoughupdates.moulconfig.common.text.StructuredText;
import org.jetbrains.annotations.ApiStatus;

import java.util.List;

/**
 * Not for manual implementation. This should be implemented by the corresponding platform.
 *
 * @see io.github.notenoughupdates.moulconfig.forge.ForgeItemStack
 */
@ApiStatus.NonExtendable
public interface IItemStack {
    List<StructuredText> getLore();

    StructuredText getDisplayName();

    int getStackSize();

    MyResourceLocation getItemId();
}

File: common/src/main/java/io/github/notenoughupdates/moulconfig/common/IItemStack.kt
package io.github.notenoughupdates.moulconfig.common

import io.github.notenoughupdates.moulconfig.common.text.StructuredText
import org.jetbrains.annotations.ApiStatus

/**
 * Not for manual implementation. This should be implemented by the corresponding platform.
 *
 * @see io.github.notenoughupdates.moulconfig.forge.ForgeItemStack
 */
@ApiStatus.NonExtendable
interface IItemStack {
    fun getLore(): List<StructuredText>
    fun getDisplayName(): StructuredText

    fun getStackSize(): Int
    fun getItemId(): MyResourceLocation
}
